# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.h
# Commit: 0552c0b4a079
# Full Hash: 0552c0b4a079a695edfca100df68c4ad82481e58
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 5
# Description:
#   Bug 1654112 - Refactor away RtcpEventObserver in favor of MediaEvents. r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D123223
# ==============================================================================

diff -r 877a0fcc6da9 -r 0552c0b4a079 dom/media/webrtc/libwebrtcglue/VideoConduit.h
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Mon Aug 16 15:57:10 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.h	Thu Aug 19 11:49:29 2021 +0200
@@ -226,11 +226,8 @@
   void CollectTelemetryData() override;
 
   void OnRtcpBye() override;
-
   void OnRtcpTimeout() override;
 
-  void SetRtcpEventObserver(mozilla::RtcpEventObserver* observer) override;
-
  private:
   // Don't allow copying/assigning.
   WebrtcVideoConduit(const WebrtcVideoConduit&) = delete;
@@ -249,6 +246,11 @@
 
   void DeliverPacket(rtc::CopyOnWriteBuffer packet, PacketType type) override;
 
+  MediaEventSource<void>& RtcpByeEvent() override { return mRtcpByeEvent; }
+  MediaEventSource<void>& RtcpTimeoutEvent() override {
+    return mRtcpTimeoutEvent;
+  }
+
   bool RequiresNewSendStream(const VideoCodecConfig& newConfig) const;
 
   mutable mozilla::ReentrantMonitor mTransportMonitor;
@@ -467,8 +469,9 @@
   // Protected by mTransportMonitor
   dom::RTCVideoFrameHistoryInternal mReceivedFrameHistory;
 
-  // Accessed only on main thread.
-  mozilla::RtcpEventObserver* mRtcpEventObserver = nullptr;
+  // Thread safe
+  MediaEventProducer<void> mRtcpByeEvent;
+  MediaEventProducer<void> mRtcpTimeoutEvent;
 };
 }  // namespace mozilla
 
